panic: nil pointer in ServerStream.Close

## Summary

`ServerStream.Close()` can panic with a nil pointer dereference when iterating `st.readers` if a session in the reader set has `setuppedTransport == nil`.

I have observed this occasionally with MediaMTX 1.17.0 and newer, after gortsplib was updated from v5.4.0 to v5.5.0+.

## Stack trace

Observed with gortsplib v5.5.1 via mediamtx v1.17.1, on RTMP publisher disconnecting while RTSP reader is still connected over TCP:

```
2026/05/06 10:53:34 INF [path test] runOnReady command stopped
2026/05/06 10:53:34 INF [path test] runOnNotReady command launched
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xf80e01]

goroutine 8531 [running]:
github.com/bluenviron/gortsplib/v5.(*ServerStream).readerSetInactiveUnsafe(0xc000615428?, 0xfd564d?)
	/go/pkg/mod/github.com/bluenviron/gortsplib/v5@v5.5.1/server_stream.go:348 +0x21
github.com/bluenviron/gortsplib/v5.(*ServerStream).Close(0xc00048b500)
	/go/pkg/mod/github.com/bluenviron/gortsplib/v5@v5.5.1/server_stream.go:133 +0x86
github.com/bluenviron/mediamtx/internal/stream.(*Stream).Close(0xc0003bc100)
	/s/internal/stream/stream.go:430 +0x6f
github.com/bluenviron/mediamtx/internal/core.(*path).setNotAvailable(0xc000352b48)
	/s/internal/core/path.go:891 +0x1d7
github.com/bluenviron/mediamtx/internal/core.(*path).executeRemovePublisher(0xc000352b48)
	/s/internal/core/path.go:945 +0x27
github.com/bluenviron/mediamtx/internal/core.(*path).doRemovePublisher(...)
	/s/internal/core/path.go:513
github.com/bluenviron/mediamtx/internal/core.(*path).runInner(0xc000352b48)
	/s/internal/core/path.go:322 +0xa75
github.com/bluenviron/mediamtx/internal/core.(*path).run(0xc000352b48)
	/s/internal/core/path.go:220 +0x2b2
created by github.com/bluenviron/mediamtx/internal/core.(*path).initialize in goroutine 9
	/s/internal/core/path.go:151 +0x52b
```

## Regression analysis

The crashing call site is new in v5.5.0. `ServerStream.Close()` was changed so that the readers loop now dereferences `ss.setuppedTransport` for every entry; v5.4.0's `Close()` only called `ss.Close()` and never touched the transport.

**v5.4.0** — https://github.com/bluenviron/gortsplib/blob/v5.4.0/server_stream.go#L138-L150

```go
func (st *ServerStream) Close() {
    st.mutex.Lock()
    st.closed = true
    st.mutex.Unlock()

    for ss := range st.readers {
        ss.Close()                      // does not touch setuppedTransport
    }
    for _, sm := range st.medias {
        sm.close()
    }
}
```

**v5.5.0+** — https://github.com/bluenviron/gortsplib/blob/v5.5.1/server_stream.go#L127-L139

```go
func (st *ServerStream) Close() {
    st.mutex.Lock()
    st.closed = true

    for ss := range st.readers {
        st.readerSetInactiveUnsafe(ss)  // NEW — dereferences ss.setuppedTransport
        st.readerRemoveUnsafe(ss)
        ss.Close()
    }
    st.mutex.Unlock()
}
```

`readerSetInactiveUnsafe` — https://github.com/bluenviron/gortsplib/blob/v5.5.1/server_stream.go#L347-L356

```go
func (st *ServerStream) readerSetInactiveUnsafe(ss *ServerSession) {
    if ss.setuppedTransport.Protocol == ProtocolUDPMulticast {   // line 348 — SIGSEGV
        ...
    } else {
        delete(st.activeUnicastReaders, ss)
    }
}
```

If a `*ServerSession` is present in `st.readers` with `setuppedTransport == nil` when `Close()` runs, the dereference at line 348 panics. This could happen due to a failed/racing SETUP or teardown path. The old `Close()` did not access this field, so this state did not trigger a panic before v5.5.0.

## Suggested fix (open to alternatives)

Either guard inside `readerSetInactiveUnsafe`:

```go
func (st *ServerStream) readerSetInactiveUnsafe(ss *ServerSession) {
    if ss.setuppedTransport == nil {
        return
    }
    ...
}
```

…or skip such sessions in the `Close()` loop. Happy to send a PR if you tell me which shape you'd prefer.


## Reproduction

No deterministic repro yet. Triggers occasionally on publisher disconnect against paths configured with `runOnReady` / `runOnNotReady`. Happens under normal traffic, not under any unusual load pattern.

## Environment

- gortsplib **v5.5.1** (via `bluenviron/mediamtx:1.17.1-ffmpeg` Docker image)
- Host: Linux x86_64

--- a/server_session.go
+++ b/server_session.go
@@ -1003,18 +1003,6 @@ func (ss *ServerSession) handleRequestInner(sc *ServerConn, req *base.Request) (
 				}, liberrors.ErrServerMediaAlreadySetup{}
 			}
 
-			if ss.state == ServerSessionStateInitial {
-				err = stream.readerAdd(ss,
-					inTH.ClientPorts,
-					protocol,
-				)
-				if err != nil {
-					return &base.Response{
-						StatusCode: base.StatusBadRequest,
-					}, err
-				}
-			}
-
 			th := headers.Transport{
 				Profile: inTH.Profile,
 			}
@@ -1076,6 +1064,50 @@ func (ss *ServerSession) handleRequestInner(sc *ServerConn, req *base.Request) (
 				} else if isSecure(inTH.Profile) {
 					srtpOutCtx = stream.medias[medi].srtpOutCtx
 				}
+
+				var mk *mikey.Message
+				mk, err = contextToMikey(srtpOutCtx)
+				if err != nil {
+					return &base.Response{
+						StatusCode: base.StatusInternalServerError,
+					}, err
+				}
+
+				var enc base.HeaderValue
+				enc, err = headers.KeyMgmt{
+					URL:          req.URL.String(),
+					MikeyMessage: mk,
+				}.Marshal()
+				if err != nil {
+					return &base.Response{
+						StatusCode: base.StatusInternalServerError,
+					}, err
+				}
+
+				// always return KeyMgmt even if redundant when playing
+				// (since it's already present in the SDP)
+				res.Header["KeyMgmt"] = enc
+			}
+
+			ss.propsMutex.Lock()
+
+			ss.setuppedTransport = &SessionTransport{
+				Protocol: protocol,
+				Profile:  inTH.Profile,
+			}
+
+			if ss.state == ServerSessionStateInitial {
+				err = stream.readerAdd(ss,
+					inTH.ClientPorts,
+				)
+				if err != nil {
+					ss.setuppedTransport = nil
+					ss.propsMutex.Unlock()
+
+					return &base.Response{
+						StatusCode: base.StatusBadRequest,
+					}, err
+				}
 			}
 
 			var udpRTPReadPort int
@@ -1127,12 +1159,7 @@ func (ss *ServerSession) handleRequestInner(sc *ServerConn, req *base.Request) (
 				th.InterleavedIDs = &[2]int{tcpChannel, tcpChannel + 1}
 			}
 
-			ss.propsMutex.Lock()
-
-			ss.setuppedTransport = &SessionTransport{
-				Protocol: protocol,
-				Profile:  inTH.Profile,
-			}
+			res.Header["Transport"] = th.Marshal()
 
 			sm := &serverSessionMedia{
 				ss:               ss,
@@ -1163,33 +1190,6 @@ func (ss *ServerSession) handleRequestInner(sc *ServerConn, req *base.Request) (
 			}
 
 			ss.propsMutex.Unlock()
-
-			res.Header["Transport"] = th.Marshal()
-
-			if isSecure(inTH.Profile) {
-				var mk *mikey.Message
-				mk, err = contextToMikey(sm.srtpOutCtx)
-				if err != nil {
-					return &base.Response{
-						StatusCode: base.StatusInternalServerError,
-					}, err
-				}
-
-				var enc base.HeaderValue
-				enc, err = headers.KeyMgmt{
-					URL:          req.URL.String(),
-					MikeyMessage: mk,
-				}.Marshal()
-				if err != nil {
-					return &base.Response{
-						StatusCode: base.StatusInternalServerError,
-					}, err
-				}
-
-				// always return KeyMgmt even if redundant when playing
-				// (since it's already present in the SDP)
-				res.Header["KeyMgmt"] = enc
-			}
 		}
 
 		return res, err

--- a/server_stream.go
+++ b/server_stream.go
@@ -203,7 +203,6 @@ func (st *ServerStream) Stats() *ServerStreamStats {
 func (st *ServerStream) readerAdd(
 	ss *ServerSession,
 	clientPorts *[2]int,
-	protocol Protocol,
 ) error {
 	st.mutex.Lock()
 	defer st.mutex.Unlock()
@@ -212,11 +211,11 @@ func (st *ServerStream) readerAdd(
 		return liberrors.ErrServerStreamClosed{}
 	}
 
-	switch protocol {
+	switch ss.setuppedTransport.Protocol {
 	case ProtocolUDP:
 		// check whether UDP ports and IP are already assigned to another reader
 		for r := range st.readers {
-			if protocol == ProtocolUDP &&
+			if r.setuppedTransport.Protocol == ProtocolUDP &&
 				r.author.ip().Equal(ss.author.ip()) &&
 				r.author.zone() == ss.author.zone() {
 				for _, rt := range r.setuppedMedias {

